WordPress/twentytwentyone
# Header: Center logo above header if all 3 items are visible

## Summary
If the logo, site title/tagline, and nav menu are all visible, the logo should be centered above the header. If the title/tagline is hidden, the logo should stay to the left.

This also incorporates changes from #362 & #406 — so it also fixes #358 & fixes #398

## Relevant technical choices:

This is using `row-gap` on `site-header` to add extra spacing between the site title/tagline & menu when the menu is long enough to wrap under the title. This allows for better vertical centering vs using a margin, which pushes the nav down. This is not supported in Safari (yet, appears to be on the way), or IE11. 

Vertical alignment with the current `margin-bottom` approach:
<img width="819" alt="Screen Shot 2020-10-12 at 3 42 38 PM" src="https://user-images.githubusercontent.com/541093/95788147-f0dd3500-0ca8-11eb-871b-d1d95086541b.png">

See below for the `row-gap` alignment in this branch.

On Safari, if you have a long menu, you don't get a space anymore:

![Screen Shot 2020-10-12 at 4 38 26 PM](https://user-images.githubusercontent.com/541093/95788476-9c868500-0ca9-11eb-83f1-fd0e31bc7b91.png)

So this is a trade-off, but IMO the improved vertical alignment outweighs the edge case of a long menu + older browser, especially since it's not unreadable.

## Test instructions

This PR can be tested by following these steps:
1. Toggle menus, site titles, and logos.
2. Check that it looks like you expect.

## Screenshots

All taken on Mac/Firefox.

1. Has a logo, menu, and title is visible

    ![screenshot-1](https://user-images.githubusercontent.com/541093/95787440-8677c500-0ca7-11eb-8ff8-db8c39198419.png)

2. Has a logo, menu, and title is not visible

    ![screenshot-2](https://user-images.githubusercontent.com/541093/95787441-87105b80-0ca7-11eb-9772-619f2c881c6b.png)

3. Has a logo, does not have a menu, and title is visible

    ![screenshot-3](https://user-images.githubusercontent.com/541093/95787442-87a8f200-0ca7-11eb-9b1d-73ddba0ab9e6.png)

4. Has a logo, does not have a menu, and title is not visible

    ![screenshot-4](https://user-images.githubusercontent.com/541093/95787444-87a8f200-0ca7-11eb-89d2-5b913d1f6b01.png)

5. No logo, has menu, and title is visible

    ![screenshot-5](https://user-images.githubusercontent.com/541093/95787445-88418880-0ca7-11eb-9607-f94257d241b3.png)

6. No logo, has menu, and title is not visible

    ![screenshot-6](https://user-images.githubusercontent.com/541093/95787446-88da1f00-0ca7-11eb-925c-53e25569df4e.png)

7. No logo, does not have a menu, and title is visible

    ![screenshot-7](https://user-images.githubusercontent.com/541093/95787447-88da1f00-0ca7-11eb-9322-912544bd3b71.png)

8. No logo, does not have a menu, and title is not visible

    ![screenshot-8](https://user-images.githubusercontent.com/541093/95787448-8972b580-0ca7-11eb-9f95-439a3652239a.png)

9. Long menu

    ![screenshot-9](https://user-images.githubusercontent.com/541093/95787449-8972b580-0ca7-11eb-9e0f-3dfc018e6d3d.png)

## Quality assurance
